# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Forecasting  
I chose forecasting because FlyRank’s search data shows clear daily fluctuations in client interactions. Anticipating these patterns can help allocate support resources more efficiently. This lane is practical, connects directly to the dataset, and has immediate decision‑making value.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

- **Decision improved:** How many staff and technical resources should be scheduled each day to handle client interactions.  
- **Who acts on it:** Operations managers and support teams who plan staffing levels and system capacity.  
- **Cost of a wrong call:**  
- **Underestimate:** Clients face delays, poor service, and higher churn risk.  
- **Overestimate:** Wasted staffing costs and idle resources that reduce efficiency.  


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Show basic stats for the 'interactions' column (replace with the actual column name if different)
df['interactions'].describe()


FileNotFoundError: [Errno 2] No such file or directory: 'data/raw/content_refresh_anonymized.csv'

In [2]:
import pandas as pd

# Correct relative path from work/notebooks/ to data/raw/
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# Show basic stats for the interactions column
df['interactions'].describe()


KeyError: 'interactions'

In [3]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# Show the first few rows and the column names
print(df.columns)
df.head()


Index(['content_id', 'client_id', 'search_volume', 'competition',
       'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count',
       'char_count', 'provider_used', 'model_used', 'impressions_90d',
       'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d',
       'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d',
       'days_with_impressions', 'days_with_sessions', 'impressions_last_30d',
       'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d',
       'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier',
       'age_tier_order', 'days_since_last_update', 'freshness_tier',
       'word_count_tier', 'char_count_tier', 'ctr', 'avg_position',
       'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier',
       'position_tier', 'trend_direction', 'trend_pct'],
      dtype='object')


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [4]:
df[['search_volume','impressions_90d','clicks_90d']].describe()


,search_volume,impressions_90d,clicks_90d
count,27532.000000,30000.000000,30000.000000
mean,158.882391,5200.366300,16.097333
std,1518.270825,16838.019547,75.076958
min,0.000000,1.000000,0.000000
25%,0.000000,81.000000,0.000000
50%,10.000000,731.000000,1.000000
75%,20.000000,3615.250000,7.000000
max,74000.000000,517715.000000,4178.000000


From the dataset:

- **Average search volume:** ~159  
- **Median impressions (90 days):** ~731  
- **Maximum clicks (90 days):** ~4,178  

These numbers show large variability in content performance. Forecasting these metrics can help anticipate demand and allocate resources more effectively. A wrong forecast could mean missing a surge in traffic or wasting effort on content that won’t perform.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**Careful words: what I can and can’t claim**

- I can claim that the dataset shows clear variability in search volume, impressions, and clicks.  
- I can claim that forecasting these metrics could help managers anticipate demand and allocate resources more effectively.  
- I cannot claim that my forecast will be perfectly accurate or guarantee business outcomes.  
- I cannot claim causation (e.g., that higher impressions directly cause more revenue) — only correlation or observed patterns.  
- I cannot claim that the dataset represents all clients or all possible scenarios; it is a sample with its own limits.  


**Self‑check**

- ✅ I chose a lane (Forecasting) that connects directly to the dataset and has clear decision‑making value.  
- ✅ I answered the framing question (decision, action, cost of a wrong call).  
- ✅ I showed 2–3 real numbers from the dataset that make the lane worth pursuing.  
- ✅ I used careful words to state what I can and cannot claim.  
- ✅ My notebook has a mix of Markdown (explanations) and Python (data exploration) cells.  


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.